kani-compiler/src/codegen_cprover_gotoc/codegen/place.rs

```diff
@@ -642,8 +642,30 @@ impl<'tcx> GotocCtx<'tcx> {
         let projection =
             unwrap_or_return_codegen_unimplemented!(self, self.codegen_place_stable(place));
         if self.use_thin_pointer_stable(place_ty) {
-            // Just return the address of the place dereferenced.
-            projection.goto_expr.address_of()
+            // For ZST objects rustc does not necessarily generate individual objects.
+            let need_not_be_unique = match projection.goto_expr.value() {
```

Comment:
I believe that's already happening, because the `match` filters for symbol expressions and not member expressions.

Reply:
Interesting... I guess you tend to write things using goto as much as possible, while I'm more familiar with MIR. The only caveat here is that this could be a member of a ZST object. In that case, we would still want the non-determinism.

Reply:
Ah, good call, thank you!